src/main/java/com/box/sdk/BoxAPIRequest.java

```diff
@@ -321,7 +321,7 @@ private BoxAPIResponse trySend(ProgressListener listener) {
         HttpURLConnection connection = this.createConnection();
 
         if (this.bodyLength > 0) {
-            connection.setFixedLengthStreamingMode(this.bodyLength);
+            connection.setFixedLengthStreamingMode((int) this.bodyLength);
```

Comment:
Wow - that's an unfortunate limitation of Java 6. We should be using chunked streaming for all uploads, so I think we should still be okay with large files.